src/HtmlTemplate.php

```diff
@@ -157,7 +157,7 @@ protected function _setOrAppend($tag, string $value = null, bool $encodeHtml = t
         // in this case we don't throw exception if tags don't exist
         if (is_array($tag) && $value === null) {
             foreach ($tag as $k => $v) {
-                $this->_setOrAppend($k, $v, $encodeHtml, $append, false);
+                $this->_setOrAppend($k, (string) $v, $encodeHtml, $append, false);
```

Comment:
But if you use `->set($model)`, then it will try to set model `id` which is integer into template too.

Reply:
see l153/typecasting above, this is already fully handled when set thru model, we cannot do it with array, as php type itself is not enought to determine DBAL/typecasting type, since #1987 we format `int` using UI persistence config, so `(string)` will not produce desired output